.golangci.yml

```diff
@@ -159,30 +159,110 @@ linters-settings:
+# - funlen            # Tool for detection of long functions.
+# - gochecknoglobals  # Check that no global variables exist.
+# - gochecknoinits    # Checks that no init functions are present in Go code.
+# - godot             # Check if comments end in a period.
+# - goerr113          # Go linter to check the errors handling expressions.
+# - gofumpt           # Gofumpt checks whether code was gofumpt-ed.
+# - gomnd             # An analyzer to detect magic numbers.
+# - inamedparam       # reports interfaces with unnamed method parameters.
+# - ireturn           # Accept Interfaces, Return Concrete Types.
+# - maintidx          # Maintidx measures the maintainability index of each function.
+# - makezero          # Finds slice declarations with non-zero initial length.
+# - maligned          # The repository of the linter has been archived by the owner. Replaced by govet 'fieldalignment'.
+# - musttag           # Enforce field tags in (un)marshaled structs.
+# - nestif            # Reports deeply nested if statements.
+# - nilnil            # Checks that there is no simultaneous return of nil error and an invalid value.
+# - nlreturn          # Nlreturn checks for a new line before return and branch statements to increase code clarity.
+# - noctx             # Finds sending http request without context.Context.
+# - nolintlint        # Reports ill-formed or insufficient nolint directives.
+# - nonamedreturns    # Reports all named returns.
+# - paralleltest      # Detects missing usage of t.Parallel() method in your Go test.
+# - predeclared       # Find code that shadows one of Go's predeclared identifiers.
+# - promlinter        # Check Prometheus metrics naming via promlint.
+# - protogetter       # Reports direct reads from proto message fields when getters should be used.
+# - rowserrcheck      # Checks whether Rows.Err of rows is checked successfully.
+# - sqlclosecheck     # Checks that sql.Rows and sql.Stmt are closed.
+# - tenv              # tenv is analyzer that detects using os.Setenv instead of t.Setenv since Go1.17
+# - testifylint       # Checks usage of github.com/stretchr/testify.
+# - thelper           # Thelper detects tests helpers which is not start with t.Helper() method.
+# - usestdlibvars     # A linter that detect the possibility to use variables/constants from the Go standard library.
+# - wastedassign      # wastedassign finds wasted assignment statements.
+# - wrapcheck         # Checks that errors returned from external packages are wrapped.
+# - wsl               # Add or remove empty lines.
+# TODO: Need to discuss/decide/remove
+# - exhaustive     # Check exhaustiveness of enum switch statements.
+# - godox          # Tool for detection of FIXME, TODO and other comment keywords.
+# - inamedparam    # reports interfaces with unnamed method parameters.
+# - interfacebloat # A linter that checks the number of methods inside an interface.
+# - lll            # Reports long lines.
```

Comment:
I am a bit torn here: on one hand I do not like those long lines- and we have them, but on the other hand having to read very long functions also does not help. I especially detest that with long signature lines as in golang you have to scroll to the end to see the return type. But putting each param on a new line also needs getting used to as it is not in line with golang formatter 😞

Reply:
long lines are always signal that you are doing something wrong. Always remember of https://en.wikipedia.org/wiki/The_Magical_Number_Seven,_Plus_or_Minus_Two